server/images.go

```diff
@@ -397,79 +343,71 @@ func CreateModel(ctx context.Context, name string, path string, fn func(resp api
 						}
 					}
 
-					newLayer, err := GetLayerWithBufferFromLayer(l)
+					newLayer, err := GetLayerWithBufferFromLayer(layer)
 					if err != nil {
 						return err
 					}
-					newLayer.From = mp.GetNamespaceRepository()
+
+					newLayer.From = modelpath.GetNamespaceRepository()
 					layers = append(layers, newLayer)
 				}
-			}
-		case "adapter":
-			fn(api.ProgressResponse{Status: fmt.Sprintf("creating model %s layer", c.Name)})
 
-			fp, err := filenameWithPath(path, c.Args)
-			if err != nil {
-				return err
+				deleteMap[manifest.Config.Digest] = struct{}{}
```

Comment:
Why is creation adding files to a deleteMap?

Reply:
`deleteMap` is a reference counter for unused layer. On creation, a layer may be unused if the new tag overrides the previous tag. Adding a layer to `deleteMap` signals the layer may be dangling after creation if it's not referenced by other models.

e.g. model A has layers a, b, c. User creates a new model from A, also called A, overriding layer b with d. Layer b is now dangling and should be removed.